api/src/main/scala/Geoprocessing.scala

```diff
@@ -48,6 +48,69 @@ trait Geoprocessing extends Utils {
   }
 
   /**
+    * For an InputData object, return a histogram of raster lines join results.
+    *
+    * @param   input  The InputData
+    * @return         A histogram of results
+    */
+  def getRasterLinesJoin(input: InputData): ResultInt = {
+    val aoi = createAOIFromInput(input)
+    val rasterLayers = cropRastersToAOI(input.rasters, input.zoom, aoi)
+    val lines = input.vector match {
+      case Some(vector) =>
+        input.vectorCRS match {
+          case Some(crs) =>
+            createMultiLineFromInput(vector, crs, input.rasterCRS)
+          case None =>
+            throw new Exception("Request data missing required 'vectorCRS'.")
+        }
+      case None =>
+        throw new Exception("Request data missing required 'vector'.")
+    }
+    ResultInt(rasterLinesJoin(rasterLayers, lines))
+  }
+
+  /**
+    * Given a collction of rasterLayers & a collection of lines, return the
+    * values intersected by the rasterized lines.
+    *
+    * @param   rasterLayers  A sequence of TileLayerCollections
+    * @param   lines         A sequence of MultiLines
+    * @return                A map of pixel counts
+    */
+  private def rasterLinesJoin(
+    rasterLayers: Seq[TileLayerCollection[SpatialKey]],
+    lines: Seq[MultiLine]
+  ): Map[String, Int] = {
+    val metadata = rasterLayers.head.metadata
+    var pixelGroups: TrieMap[(List[Int], List[Int], SpatialKey), Int] = TrieMap.empty
```

Comment:
Probably worth replacing the second and third parts of TrieMap key -- `(List[Int], List[Int], SpatialKey)` -- here with a tuple of `(Int, Int, SpatialKey)` representing the column, row, and tile key.

Reply:
Here's a suggested implementation:

```diff
diff --git a/api/src/main/scala/Geoprocessing.scala b/api/src/main/scala/Geoprocessing.scala
index 3fb8e3e..32667a1 100644
--- a/api/src/main/scala/Geoprocessing.scala
+++ b/api/src/main/scala/Geoprocessing.scala
@@ -70,6 +70,8 @@ trait Geoprocessing extends Utils {
     ResultInt(rasterLinesJoin(rasterLayers, lines))
   }
 
+  private case class TilePixel (key: SpatialKey, col: Int, row: Int)
+
   /**
     * Given a collction of rasterLayers & a collection of lines, return the
     * values intersected by the rasterized lines.
@@ -83,7 +85,7 @@ trait Geoprocessing extends Utils {
     lines: Seq[MultiLine]
   ): Map[String, Int] = {
     val metadata = rasterLayers.head.metadata
-    var pixelGroups: TrieMap[(List[Int], List[Int], SpatialKey), Int] = TrieMap.empty
+    var pixelGroups: TrieMap[(List[Int], TilePixel), Int] = TrieMap.empty
 
     joinCollectionLayers(rasterLayers).par
       .foreach({ case (key, tiles) =>
@@ -93,8 +95,8 @@ trait Geoprocessing extends Utils {
 
         lines.par.foreach({ multiLine =>
           Rasterizer.foreachCellByMultiLineString(multiLine, re) { case (col, row) =>
-            val pixelGroup: (List[Int], List[Int], SpatialKey) =
-              (tiles.map(_.get(col, row)).toList, List(col, row), key)
+            val pixelGroup: (List[Int], TilePixel) =
+              (tiles.map(_.get(col, row)).toList, TilePixel(key, col, row))
             pixelGroups.getOrElseUpdate(pixelGroup, 1)
           }
         })
@@ -102,7 +104,7 @@ trait Geoprocessing extends Utils {
 
     pixelGroups
       .toSeq
-      .map { case ((key, _, _), value) => (key, value) }
+      .map { case ((key, _), value) => (key, value) }
       .groupBy(_._1)
       .map { case (k, v) =>
         k.toString -> v.map { case (_, n) => n }.sum
```